app.js

```diff
@@ -207,30 +207,41 @@ function orchestrate(spec, isManifest, token) {
+                                                    exec(cmd, function (err, stdout, stderr) {
+                                                      if (err) console.error(stderr);
+                                                    });
+                                                    spec.history = spec.history.add('The document has been published at <a href="' +
+                                                        report.metadata.get('thisVersion') + '">' + report.metadata.get('thisVersion') + '</a>.');
+                                                    return Promise.resolve("finished");
+                                                }, function (err) {
+                                                    spec.jobs['update-tr-shortlink'].status = 'error';
+                                                    spec.jobs['update-tr-shortlink'].errors.push(err.toString());
+                                                    return Promise.reject(err);
                                                 });
-                                                spec.history = spec.history.add('The document has been published at <a href="' + report.metadata.get('thisVersion') + '">' + report.metadata.get('thisVersion') + '</a>.');
-                                                return Promise.resolve("finished");
                                             }, function (err) {
-                                                spec.jobs['update-tr-shortlink'].status = 'error';
-                                                spec.jobs['update-tr-shortlink'].errors.push(err.toString());
+                                                spec.jobs['tr-install'].status = 'error';
+                                                spec.jobs['tr-install'].errors.push(err.toString());
                                                 return Promise.reject(err);
                                             });
-                                        }, function (err) {
-                                            spec.jobs['tr-install'].status = 'error';
-                                            spec.jobs['tr-install'].errors.push(err.toString());
-                                            return Promise.reject(err);
-                                        });
+                                        }
+                                        else {
+                                            spec.jobs['publish'].status = 'failure';
+                                            spec.jobs['publish'].errors.push(errors);
+                                            spec.history = spec.history.add('The document could not be published: ' + errors.map(function (error) {
+                                                return error.message;
+                                            }));
+                                            return Promise.reject(new Error("Failed the publication system"));
+                                        }
```

Comment:
One caveat here: if we are entering this `else` block (from line 237 to line 244), then `if(errors.length === 0)` (defined at line 206) still applies, so you will always get the following history message: `The document could not be published: `.

Not sure what you should display instead as I was wondering the purpose of the check in my comment above.